server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -1327,12 +1328,16 @@ public ClusterState execute(ClusterState currentState) {
                     request.indexSettings(),
                     request.ignoreIndexSettings()
                 );
-                if (snapshotIndexMetadata.getCompatibilityVersion().before(minIndexCompatibilityVersion)) {
+                if (snapshotIndexMetadata.getCompatibilityVersion().isLegacyIndexVersion()) {
```

Comment:
Similarly, it does not have an impact on 8.x but makes the code look similar to main.